leptos_macro/src/component.rs

```diff
@@ -407,106 +365,43 @@ impl Docs {
-    Optional,
-    OptionalNoStrip,
-    OptionalWithDefault(syn::Lit),
-    StripOption,
-    Into,
-}
-
-impl PropOpt {
-    fn from_attribute(attr: &Attribute) -> Option<HashSet<Self>> {
-        const ABORT_OPT_MESSAGE: &str =
-            "only `optional`, `optional_no_strip`, `strip_option`, `default` \
-             and `into` are allowed as arguments to `#[prop()]`";
-
-        if attr.path != parse_quote!(prop) {
-            return None;
-        }
-
-        if let Meta::List(MetaList { nested, .. }) =
-            attr.parse_meta().unwrap_or_abort()
-        {
-            Some(
-                nested
-                    .iter()
-                    .map(|opt| match opt {
-                        NestedMeta::Meta(Meta::Path(opt)) => {
-                            if *opt == parse_quote!(optional) {
-                                PropOpt::Optional
-                            } else if *opt == parse_quote!(optional_no_strip) {
-                                PropOpt::OptionalNoStrip
-                            } else if *opt == parse_quote!(strip_option) {
-                                PropOpt::StripOption
-                            } else if *opt == parse_quote!(into) {
-                                PropOpt::Into
-                            } else {
-                                abort!(
-                                    opt,
-                                    "invalid prop option";
-                                    help = ABORT_OPT_MESSAGE
-                                );
-                            }
-                        }
-                        NestedMeta::Meta(Meta::NameValue(MetaNameValue {
-                            path,
-                            eq_token: _,
-                            lit,
-                        })) => {
-                            if *path == parse_quote!(default) {
-                                PropOpt::OptionalWithDefault(lit.to_owned())
-                            } else {
-                                abort!(
-                                    opt,
-                                    "invalid prop option";
-                                    help = ABORT_OPT_MESSAGE
-                                );
-                            }
-                        }
-                        _ => abort!(opt, ABORT_OPT_MESSAGE,),
-                    })
-                    .collect(),
-            )
-        } else {
-            abort!(
-                attr,
-                "the syntax for `#[prop]` is incorrect";
-                help = "try `#[prop(optional)]`";
-                help = ABORT_OPT_MESSAGE
-            );
-        }
-    }
+#[derive(Clone, Debug, AttributeDerive)]
+#[attribute(ident = prop)]
+struct PropOpt {
+    #[attribute(conflicts = [optional_no_strip, strip_option])]
+    optional: bool,
+    #[attribute(conflicts = [optional, strip_option])]
+    optional_no_strip: bool,
+    #[attribute(conflicts = [optional, optional_no_strip])]
```

Comment:
these are not needed as already implied by the other two fields, just added them for completenes.